### PR TITLE
So much change! So much perf!

### PR DESCRIPTION
**Pull request by joeybaker**
* dramatic reduction of resources by not watching node_modules
* dramatic speed up of watching by using chokidar instead of gaze which gets us native OS FS events were possible
* no longer running three file watchers, sometimes with overlapping files. We should have a much smaller memory and CPU footprint.
* enables setting the HTML that will be served at `/default` by either file or function. Super useful if you want to server-render.
* unifed logging. It's all pretty now.

Breaking changes:
* we no longer watch all css files under the cwd. Only the entry file and it's dependencies
* I removed a bunch of code added by @serapath in #46. @serapath if you can try this branch and confirm for me that nothing is broken? Fortunately, the APIs that were removed are undocumented.

### CODE REVIEW

**Comment by joeybaker** [comment]
@serapath @atomify/owners can you guys please take a look at this. It contains (hopefully minor) breaking changes, but it (anecdotally) has some pretty major perf improvements to livereloading.

**Comment by serapath** [comment]
The tutorial from the [slides](http://serapath.github.io/atomify) still works, so i guess its ok :-)

**Comment by joeybaker** [comment]
@serapath great!!

Also, I will update this in a it - I want to change the HTML API to be asynchronous.

**Comment by joeybaker** [comment]
Okay… I'm pretty happy with this now. I'd love someone from @atomify/owners to look at it though :)

File: lib/server.js [hunk 1]
```diff
@@ -65,9 +34,61 @@ internals.serveDefaultPage = function serveDefaultPage (res, args) {
...
-  res.setHeader('Content-Type', 'text/html')
+  return src
+}
+
+internals.customDefaultHTML = ''
+
+internals.setCustomDefaultHTML = function setCustomDefaultHTML (file, args){
+  var html = fs.readFileSync(file, {encoding: 'utf8'})
+    , shouldReplaceJS = ~html.indexOf(JS_REPLACE)
+    , shouldReplaceCSS = ~html.indexOf(CSS_REPLACE)
```
Comment (bclinkinbeard) [inline lib/server.js]: What does `~` do? I've never seen that before.

Comment (joeybaker) [inline lib/server.js]: Oh crap, sorry. That's probably not good in production code. It's hard to read. 

`~` is a bit shift operator. so…

```js
~-1 === 0
```

It's just a shortcut for `html.indexOf(JS_REPLACE) > -1`

**Comment by Techwraith** [comment]
LGTM. Click that merge button :)

**Pull request merged**